### PR TITLE
[pulsar-admin] allow create functions with package URL

### PR DESCRIPTION
**Pull request by freeznet**
Fix https://github.com/apache/pulsar/issues/11665

### Motivation

Allow user to create function with package URL with pulsar-admin.

### Modifications

- allow passing valid package URL from pulsar-admin functions
- added tests

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

#### For contributor

For this PR, do we need to update docs?

- If yes, please update docs or create a follow-up issue if you need help.
  
- If no, please explain why.

#### For committer

For this PR, do we need to update docs?

- If yes,
  
  - if you update docs in this PR, label this PR with the `doc` label.
  
  - if you plan to update docs later, label this PR with the `doc-required` label.

  - if you need help on updating docs, create a follow-up issue with the `doc-required` label.
  
- If no, label this PR with the `no-need-doc` label and explain why.

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@freeznet thanks for your contribution! When can we start updating docs?

**Comment by freeznet** [comment]
@Anonymitaet thanks, we can start the doc once this PR is merged.

**Review by zymap — APPROVED** [review]

File: pulsar-common/src/main/java/org/apache/pulsar/common/functions/Utils.java [hunk 1]
```diff
@@ -34,7 +36,9 @@
 
     public static boolean isFunctionPackageUrlSupported(String functionPkgUrl) {
         return isNotBlank(functionPkgUrl) && (functionPkgUrl.startsWith(HTTP)
-                || functionPkgUrl.startsWith(FILE));
+                || functionPkgUrl.startsWith(FILE)
+                || Arrays.stream(PackageType.values()).anyMatch(type -> functionPkgUrl.startsWith(type.toString()))
```
Comment (eolivelli) [inline pulsar-common/src/main/java/org/apache/pulsar/common/functions/Utils.java:40]: can we create an utility method ? this code looks duplicated

**Review by eolivelli — CHANGES_REQUESTED** [review]
overall looks good

please take a look to my comment about reducing duplicated code

**Review by eolivelli — APPROVED** [review]
LGTM

**Review by nlu90 — APPROVED** [review]
LGTM

**Comment by Anonymitaet** [comment]
> @Anonymitaet thanks, we can start the doc once this PR is merged.

@freeznet please ping me if you submit a doc PR, thanks

**Review by sijie — APPROVED** [review]

**Pull request merged**